systemd/system/oem-cloudinit.service

```
@@ -0,0 +1,11 @@
+[Unit]
+Description=Run cloudinit
+
+[Service]
+EnvironmentFile=/var/run/ignition.env
+Type=oneshot
+ExecCondition=/usr/bin/bash -xc 'OEMS=(ec2-compat gce rackspace-onmetal azure cloudsigma packet vmware digitalocean); [[ "$${OEMS[*]}" =~ "${OEM_ID}" ]]'
+ExecStart=/usr/bin/coreos-cloudinit --oem="${OEM_ID}"
```

This here should use `ec2-compat` as argument for `coreos-cloudinit` and not `ec2`

---

With Ignition v3 the OEM_ID value inside the `/var/run/ignition.env` file will be `aws`, I suggest to put `aws` in the OEMS list and then change the ExecStart line to pass `ec2-compat` in case `aws` is the OEM_ID value, and otherwise pass it as is.

---

```suggestion
ExecCondition=/usr/bin/bash -xc 'OEMS=(aws gce rackspace-onmetal azure cloudsigma packet vmware digitalocean); echo $${OEMS[*]} | tr ' ' '\n' | grep -q -x -F "$${OEM_ID}"'
ExecStart=/usr/bin/bash -xc '/usr/bin/coreos-cloudinit --oem="$(if [ "$${OEM_ID}" = aws ]; then echo ec2-compat; else ; echo "$${OEM_ID}" ; fi)"'
```

I also noticed that the `=~` matching also worked for substrings, here I used grep to match the full string of the entry.